bigtable/bttest/inmem.go

```diff
@@ -197,10 +201,59 @@ func (s *server) DeleteTable(ctx context.Context, req *btapb.DeleteTableRequest)
 	if _, ok := s.tables[req.Name]; !ok {
 		return nil, status.Errorf(codes.NotFound, "table %q not found", req.Name)
 	}
+	if s.tables[req.Name].isProtected {
+		return nil, status.Errorf(codes.FailedPrecondition, "table %q is protected from deletion", req.Name)
+	}
 	delete(s.tables, req.Name)
 	return &emptypb.Empty{}, nil
 }
 
+func (s *server) UpdateTable(ctx context.Context, req *btapb.UpdateTableRequest) (*longrunning.Operation, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	tbl, ok := s.tables[req.GetTable().GetName()]
+	if !ok {
+		return nil, status.Errorf(codes.NotFound, "table %q not found", req.GetTable().GetName())
+	}
+	tbl.isProtected = req.GetTable().GetDeletionProtection()
```

Comment:
We use [update_mask](https://source.corp.google.com/piper///depot/google3/google/bigtable/admin/v2/bigtable_table_admin.proto;bpv=1;bpt=1;l=801?gsn=update_mask&gs=kythe%3A%2F%2Fgoogle3%3Flang%3Dprotobuf%3Fpath%3Dgoogle%2Fbigtable%2Fadmin%2Fv2%2Fbigtable_table_admin.proto%234.9.2.1)